channeldb/invoices.go

```diff
@@ -2112,13 +2112,22 @@ func (d *DB) addHTLCs(invoices, settleIndex, //nolint:funlen
 		if settleEligibleAMP {
 			htlcContextState = invpkg.ContractSettled
 		}
-		htlcSettled, err := updateHtlc(
-			timestamp, htlc, htlcContextState, setID,
+
+		htlcStateChanged, htlcState, err := getUpdatedHtlcState(
+			htlc, htlcContextState, setID,
 		)
 		if err != nil {
 			return nil, err
 		}
 
+		if htlcStateChanged {
+			htlc.State = htlcState
+			htlc.ResolveTime = timestamp
```

Comment:
Ah yeah resolve time doesn't need to be passed in anymore, as we already know it before we call, and before it was just there so we could modify the pointer passed in.

Reply:
Also still getting through the set of commits, but in the future, this can be put into the `ApplyUpdate` type call back.